File: core/entity.go
package core

import (
	"github.com/louisevanderlith/husk/validation"
	"github.com/louisevanderlith/kong/prime"
)

type Entity struct {
	Name           string `hsk:"size(30)"`
	ProfileKey     string
	User           prime.User
	Identification string    `hsk:"size(30)"` //This can be a Company Registration, RSA ID, Passport or another internal identifier
	Addresses      []Address `hsk:"null"`
}

func (e Entity) Valid() error {
	return validation.Struct(e)
}

File: lib/bodies/entity.dart
import 'address.dart';
import 'user.dart';

class Entity {
  final String name;
  final String profileKey;
  final User user;
  final String identification;
  final List<Address> addresses;

  Entity(this.name, this.profileKey, this.user, this.identification,
      this.addresses);

  Map<String, dynamic> toJson() {
    return {
      "Name": this.name,
      "ProfileKey": this.profileKey,
      "User": this.user,
      "Identification": this.identification,
      "Addresses": this.addresses
    };
  }
}
